src/strings/nfa-noodlification.cc

```diff
@@ -50,11 +49,59 @@ void unify_initial_and_final_states(const std::vector<std::shared_ptr<Nfa>>& nfa
 Nfa concatenate_with(const std::vector<std::shared_ptr<Nfa>>& nfas, mata::Symbol delimiter) {
     Nfa concatenation{*nfas[0]};
     for (std::vector<std::shared_ptr<Nfa>>::size_type i = 1; i < nfas.size(); ++i) {
-        concatenation = concatenate_eps(concatenation, *nfas[i], delimiter, true);
+        concatenation = mata::nfa::algorithms::concatenate_eps(concatenation, *nfas[i], delimiter, true);
     }
     return concatenation;
 }
 
+/**
+ * @brief Check if transducer @p nft T is for sure homomorphic, that is whether for each two words u,v, it holds T(u.v) = T(u).T(v).
+ * 
+ * The check is only a heuristic, it can return false even if T is homomorphic. We do this by checking whether there is exactly one
```

Comment:
Maybe warn about this using `@warning The check ... is homomorphic.\n\n We...`

Reply:
Done